File: web3-libs/mod2/playground/aula2/convertWei.js
// 1. Instale e importe a biblioteca Web3.js
import Web3 from "web3";
import readline from "readline";

// 2. Configure o provider
const RPC_URL = "http://127.0.0.1:8545";
const web3 = new Web3(RPC_URL);

// 3. Obtenha o endereço da conta via readline input
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

rl.question("Digite o endereço da conta: ", (address) => {
  web3.eth
    .getBalance(address)
    .then((balance) => {
      const etherBalance = web3.utils.fromWei(balance, "ether");
      console.log("Saldo em wei:", balance);
      console.log("Saldo em ether:", etherBalance);
    })
    .catch((error) => {
      console.error("Erro ao consultar o saldo:", error);
    });
  rl.close();
});

File: web3-libs/mod3/playground/aula2/convert_wei.py
# 1. Instale e importe a biblioteca web3.py
from web3 import Web3

# 2. Configure o provider
RPC_URL = "http://127.0.0.1:8545"
web3 = Web3(Web3.HTTPProvider(RPC_URL))

# 3. Verifique a conexão
try:
    block_number = web3.eth.get_block_number()
    print(f"Número do bloco: {block_number}")
except Exception as error:
    print("Erro ao verificar a conexão:", error)

# 4. Obtenha o endereço da conta via input
address = input("Digite o endereço da conta: ")
try:
    balance = web3.eth.get_balance(address)
    print("Saldo em wei:", balance)
    print("Saldo em ether:", web3.from_wei(balance, "ether"))
except Exception as error:
    print("Erro ao consultar o saldo:", error) 